lib/encode.ex

```diff
@@ -204,6 +205,46 @@ defmodule Jason.Encode do
     end
   end
 
+  defp map_camel_case(value, escape, encode_map) do
+    case Map.to_list(value) do
+      [] -> "{}"
+      [{key, value} | tail] ->
+        key = camelize(key)
+        ["{\"", key(key, escape), "\":",
+         value(value, escape, encode_map)
+         | map_camel_case_loop(tail, escape, encode_map)]
+    end
+  end
+
+  defp map_camel_case_loop([], _escape, _encode_map) do
+    '}'
+  end
+
+  defp map_camel_case_loop([{key, value} | tail], escape, encode_map) do
+    key = camelize(key)
```

Comment:
I think that splitting the last expression seems a bit complicated to "parse", how about this way?
```suggestion
    key = key
      |> camelize()
      |> key(escape)
    value = value(value, escape, encode_map)

    ["{\"", key, "\":", value | map_camel_case_loop(tail, escape, encode_map)]
```

Reply:
Again, the implementation you suggest looks good. I followed the way of composing the functions, affecting as few lines of code and logic as possible. We can go on with the suggestion you made.